module/VuFindSearch/src/VuFindSearch/Service.php

```diff
@@ -182,92 +182,105 @@ public function getIds(
+    ) {
+        $command = new RetrieveBatchCommand($backendId, $ids, $params);
         return $this->legacyInvoke($command, ['ids' => $ids]);
     }
 
     /**
      * Retrieve a random batch of records.
      *
-     * @param string              $backend Search backend identifier
-     * @param Query\AbstractQuery $query   Search query
-     * @param int                 $limit   Search limit
-     * @param ParamBag            $params  Search backend parameters
+     * @param string              $backendId Search backend identifier
+     * @param Query\AbstractQuery $query     Search query
+     * @param int                 $limit     Search limit
+     * @param ?ParamBag           $params    Search backend parameters
      *
      * @return RecordCollectionInterface
      *
      * @deprecated Use Service::invoke(RandomCommand $command) instead
      */
-    public function random($backend, $query, $limit = 20, $params = null)
-    {
-        $command = new RandomCommand($backend, $query, $limit, $params);
+    public function random(
+        string $backendId,
+        Query\AbstractQuery $query,
+        int $limit = 20,
+        ParamBag $params = null
+    ) {
+        $command = new RandomCommand($backendId, $query, $limit, $params);
         return $this->legacyInvoke($command, ['query' => $query, 'limit' => $limit]);
     }
 
     /**
      * Return similar records.
      *
-     * @param string   $backend Search backend identifier
-     * @param string   $id      Id of record to compare with
-     * @param ParamBag $params  Search backend parameters
+     * @param string    $backendId Search backend identifier
+     * @param string    $id        Id of record to compare with
+     * @param ?ParamBag $params    Search backend parameters
      *
      * @return RecordCollectionInterface
      *
      * @deprecated Use Service::invoke(SimilarCommand $command) instead
      */
-    public function similar($backend, $id, ParamBag $params = null)
-    {
-        $command = new SimilarCommand($backend, $id, $params);
+    public function similar(
+        string $backendId,
+        string $id,
+        ParamBag $params = null
+    ) {
+        $command = new SimilarCommand($backendId, $id, $params);
         return $this->legacyInvoke($command, ['id' => $id]);
     }
 
     /**
      * Return records for work expressions.
      *
-     * @param string   $backend  Search backend identifier
-     * @param string   $id       Id of record to compare with
-     * @param array    $workKeys Work identification keys (optional; retrieved from
-     * the record to compare with if not specified)
-     * @param ParamBag $params   Search backend parameters
+     * @param string    $backendId Search backend identifier
+     * @param string    $id        Id of record to compare with
+     * @param ?array    $workKeys  Work identification keys (optional; retrieved
+     *                             from the record to compare with if not specified)
```

Comment:
You don't need to align wrapped comments; they can go all the way left. It doesn't really hurt here, but going to the left margin as a general rule prevents things from getting too cramped when the types and parameter names are very long.

Reply:
Ah, I see `PEAR.Commenting.FunctionComment.ParamCommentAlignment` has been excluded. I have to fix it so that my IDE picks that up.